pkg/capabilities/v2/chain-capabilities/evm/proto_helpers.go

```diff
@@ -171,141 +169,193 @@ func ConvertCallMsgToProto(msg *evmtypes.CallMsg) (*CallMsg, error) {
 		Name:          filter.Name,
 		RetentionTime: int64(filter.Retention),
-		Addresses:     convertAddressesToProto(filter.Addresses),
-		EventSigs:     convertHashesToProto(filter.EventSigs),
-		Topic2:        convertHashesToProto(filter.Topic2),
-		Topic3:        convertHashesToProto(filter.Topic3),
-		Topic4:        convertHashesToProto(filter.Topic4),
+		Addresses:     evm.ConvertAddressesToProto(filter.Addresses),
+		EventSigs:     evm.ConvertHashesToProto(filter.EventSigs),
+		Topic2:        evm.ConvertHashesToProto(filter.Topic2),
+		Topic3:        evm.ConvertHashesToProto(filter.Topic3),
+		Topic4:        evm.ConvertHashesToProto(filter.Topic4),
 		MaxLogsKept:   filter.MaxLogsKept,
 		LogsPerBlock:  filter.LogsPerBlock,
 	}
 }
 
 func ConvertLPFilterFromProto(protoFilter *LPFilter) (evmtypes.LPFilterQuery, error) {
 	if protoFilter == nil {
-		return evmtypes.LPFilterQuery{}, errEmptyFilter
+		return evmtypes.LPFilterQuery{}, evm.ErrEmptyFilter
+	}
+
+	var addresses []evmtypes.Address
+	for i, protoAddress := range protoFilter.GetAddresses() {
+		address, err := evm.ConvertOptionalAddressFromProto(protoAddress)
+		if err != nil {
+			return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert address[%d]: %w", i, err)
+		}
+		addresses = append(addresses, address)
+	}
+
+	sigs, err := evm.ConvertHashesFromProto(protoFilter.GetEventSigs())
+	if err != nil {
+		return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert event sigs: %w", err)
+	}
+
+	t2, err := evm.ConvertHashesFromProto(protoFilter.GetTopic2())
+	if err != nil {
+		return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert topic2: %w", err)
+	}
+
+	t3, err := evm.ConvertHashesFromProto(protoFilter.GetTopic3())
+	if err != nil {
+		return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert topic3: %w", err)
+	}
+
+	t4, err := evm.ConvertHashesFromProto(protoFilter.GetTopic4())
+	if err != nil {
+		return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert topic4: %w", err)
 	}
 
 	return evmtypes.LPFilterQuery{
 		Name:         protoFilter.GetName(),
 		Retention:    time.Duration(protoFilter.GetRetentionTime()),
-		Addresses:    ConvertAddressesFromProto(protoFilter.GetAddresses()),
-		EventSigs:    ConvertHashesFromProto(protoFilter.GetEventSigs()),
-		Topic2:       ConvertHashesFromProto(protoFilter.GetTopic2()),
-		Topic3:       ConvertHashesFromProto(protoFilter.GetTopic3()),
-		Topic4:       ConvertHashesFromProto(protoFilter.GetTopic4()),
+		Addresses:    addresses,
+		EventSigs:    sigs,
+		Topic2:       t2,
+		Topic3:       t3,
+		Topic4:       t4,
 		MaxLogsKept:  protoFilter.GetMaxLogsKept(),
 		LogsPerBlock: protoFilter.GetLogsPerBlock(),
 	}, nil
 }
 
-func ConvertFilterToProto(filter evmtypes.FilterQuery) *FilterQuery {
+func ConvertFilterToProto(filter evmtypes.FilterQuery) (*FilterQuery, error) {
+	topics, err := convertTopicsToProto(filter.Topics)
+	if err != nil {
+		return nil, errors.Join(evm.ErrTopicsConversion, err)
```

Comment:
nit: errors.Join is used to join two independent errors. In this case, we add more context to the original, so we should use wrap.